### PR TITLE
refactor(opsem): split mem mgr templates into hdr and src

### PR DESCRIPTION
**Pull request by priyasiddharth**

### CODE REVIEW

File: lib/seahorn/BvOpSem2ExtraWideMemMgr.hh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #include "BvOpSem2Context.hh"
 #include "BvOpSem2RawMemMgr.hh"
 #include "BvOpSem2TrackingRawMemMgr.hh"
-#include "BvOpSem2WideMemManagerMixin.hh"
+#include "BvOpSem2WideMemManagerMixin.cc"
```
Comment (agurfinkel) [inline lib/seahorn/BvOpSem2ExtraWideMemMgr.hh:6]: looks like a bug. You should never include a `cc` file in a header file!

File: lib/seahorn/BvOpSem2WideMemMgr.cc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #include "BvOpSem2Context.hh"
 #include "BvOpSem2RawMemMgr.hh"
 
-#include "BvOpSem2WideMemManagerMixin.hh"
+#include "BvOpSem2WideMemManagerMixin.cc"
```
Comment (agurfinkel) [inline lib/seahorn/BvOpSem2WideMemMgr.cc:4]: `cc` file inclusion

**Comment by agurfinkel** [comment]
I am guessing this is in reaction to my request to not have so much code in the header files. But this does not follow the methodology I suggested, instead moves code into `cc` files and includes `cc` files, thus making them header files.

The end result is now worse than before -- code is still in header files, yet now the header files are hidden as `cc` files. Maybe we should talk before another big code change.

File: lib/seahorn/BvOpSem2ExtraWideMemMgr.hh (hunk 1 above)
Comment (priyasiddharth) [reply lib/seahorn/BvOpSem2ExtraWideMemMgr.hh]: fixed

File: lib/seahorn/BvOpSem2WideMemMgr.cc (hunk 2 above)
Comment (priyasiddharth) [reply lib/seahorn/BvOpSem2WideMemMgr.cc]: fixed

**Comment by priyasiddharth** [comment]
Fixed following https://isocpp.org/wiki/faq/templates#separate-template-class-defn-from-decl

**Pull request merged**